src/main.rs

```diff
@@ -132,33 +143,25 @@ pub async fn main() -> Result<(), anyhow::Error> {
         }
     };
 
-    let mut module_config = match bindle {
-        Some(name) => wagi::load_bindle(
-            name,
-            bindle_server.as_str(),
-            cache_config_path.clone(),
-            mc.clone(),
-            &log_dir,
-        )
-        .await
-        .map_err(|e| anyhow::anyhow!("Failed to load bindle {}: {}", name, e))?,
-        None => wagi::load_modules_toml(
-            module_config_path.as_str(),
-            cache_config_path.clone(),
-            mc.clone(),
-            &log_dir,
-        )
-        .await
-        .map_err(|e| anyhow::anyhow!("Failed to load TOML {}: {}", module_config_path, e))?,
+    let env_vars: HashMap<String, String> = match matches.values_of("env_vars") {
+        Some(v) => v
+            .into_iter()
+            .map(parse_env_var)
+            .collect::<anyhow::Result<_>>()?,
+        None => HashMap::new(),
     };
 
-    if module_config.default_host.is_none() {
-        log::info!("Setting default host to {}", default_host.as_str());
-        module_config.default_host = Some(default_host);
-    }
+    let builder = builder
```

Comment:
Okay, so at first I was gonna point out that really the builder thing should have gone in its own PR... but I seriously LOVE how much more elegant this is than the old way... and that really shows when you do the `match` on line 161. SOOO Much cleaner.

Reply:
Yeah, I was going to do it in its own PR, but I would have had to plumb things down the whole stack so I just did it now. Glad there is some ❤️ for this